diff --git a/feeds/packages/lang/rust/Makefile b/feeds/packages/lang/rust/Makefile
index 0afa39ca8..e22e1d97e 100644
--- a/feeds/packages/lang/rust/Makefile
+++ b/feeds/packages/lang/rust/Makefile
@@ -5,12 +5,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=rust
-PKG_VERSION:=1.80.1
+PKG_VERSION:=1.85.0
 PKG_RELEASE:=1
 
 PKG_SOURCE:=rustc-$(PKG_VERSION)-src.tar.gz
 PKG_SOURCE_URL:=https://static.rust-lang.org/dist/
-PKG_HASH:=2c0b8f643942dcb810cbcc50f292564b1b6e44db5d5f45091153996df95d2dc4
+PKG_HASH:=2f4f3142ffb7c8402139cfa0796e24baaac8b9fd3f96b2deec3b94b4045c6a8a
 HOST_BUILD_DIR:=$(BUILD_DIR)/host/rustc-$(PKG_VERSION)-src
 
 PKG_MAINTAINER:=Luca Barbato <lu_zero@luminem.org>
diff --git a/feeds/packages/lang/rust/patches/0001-Update-xz2-and-use-it-static.patch b/feeds/packages/lang/rust/patches/0001-Update-xz2-and-use-it-static.patch
index 4971f3ff2..c9bc0fddd 100644
--- a/feeds/packages/lang/rust/patches/0001-Update-xz2-and-use-it-static.patch
+++ b/feeds/packages/lang/rust/patches/0001-Update-xz2-and-use-it-static.patch
@@ -9,7 +9,7 @@ Subject: [PATCH] Update xz2 and use it static
 
 --- a/src/bootstrap/Cargo.toml
 +++ b/src/bootstrap/Cargo.toml
-@@ -62,7 +62,7 @@ tar = "0.4"
+@@ -60,7 +60,7 @@ tar = "0.4"
  termcolor = "1.4"
  toml = "0.5"
  walkdir = "2.4"
diff --git a/feeds/packages/lang/rust/rust-values.mk b/feeds/packages/lang/rust/rust-values.mk
index d423d0a87..035a4e077 100644
--- a/feeds/packages/lang/rust/rust-values.mk
+++ b/feeds/packages/lang/rust/rust-values.mk
@@ -104,3 +104,5 @@ CARGO_PKG_CONFIG_VARS= \
 	TARGET_CFLAGS="$(TARGET_CFLAGS) $(RUSTC_CFLAGS)"
 
 CARGO_PKG_PROFILE:=$(if $(CONFIG_DEBUG),dev,release)
+
+CARGO_RUSTFLAGS+=-Clink-arg=-fuse-ld=$(TARGET_LINKER)
